example/mailto/mailto.cpp

```diff
@@ -0,0 +1,377 @@
+    cc(std::size_t idx) const
+    {
+        return encoded_cc(idx).decode();
+    }
+
+    /// @copydoc cc()
+    urls::pct_string_view
+    encoded_cc(std::size_t idx) const noexcept
+    {
+        // Look for ith email address in one of the "to" headers
+        auto ps = u_.encoded_params();
+        auto it = ps.find("cc", urls::ignore_case);
+        while (it != ps.end())
+        {
+            auto s = ith_address((*it++).value, idx);
+            if (s)
+                return *s;
+            it = ps.find(it, "cc", urls::ignore_case);
+        }
+        return {};
+    }
+
+    /// Return number of "cc" email addresses in the URL
+    std::size_t
+    size_cc() const noexcept
+    {
+        // Count addresses in "to" headers
+        std::size_t n = 0;
+        auto ps = u_.encoded_params();
+        auto it = ps.find("cc", urls::ignore_case);
+        while (it != ps.end())
+        {
+            n += addr_in_str((*it++).value);
+            it = ps.find(it, "cc", urls::ignore_case);
+        }
+        return n;
+    }
+
+    /// Return email message subject
+    std::string
+    subject() const
+    {
+        return encoded_subject().decode();
+    }
+
+    /// @copydoc subject()
+    urls::pct_string_view
+    encoded_subject() const noexcept
+    {
+        return param_or_empty("subject");
+    }
+
+    /// Return email message body
+    std::string
+    body() const
+    {
+        return encoded_body().decode();
+    }
+
+    /// @copydoc body()
+    urls::pct_string_view
+    encoded_body() const noexcept
+    {
+        return param_or_empty("body");
+    }
+
+    friend
+    std::ostream&
+    operator<<(std::ostream& os, mailto_view m)
+    {
+        return os << m.u_;
+    }
+
+private:
+    // Count number of addresses in a string
+    static
+    std::size_t
+    addr_in_str(urls::string_view s)
+    {
+        std::size_t n = 0;
+        bool empty = true;
+        for (char c : s)
+        {
+            if (c == ',')
+            {
+                n += !empty;
+                empty = true;
+            }
+            else
+            {
+                empty = false;
+            }
+        }
+        n += !empty;
+        return n;
+    };
+
+    // Get the ith address from a string
+    static
+    urls::optional<urls::pct_string_view>
+    ith_address(urls::string_view to, std::size_t &idx) noexcept
+    {
+        auto p = to.find(',');
+        while (p != urls::string_view::npos)
+        {
+            if (idx == 0)
+                return urls::pct_string_view(
+                    to.substr(0, p));
+            --idx;
+            to.remove_prefix(p + 1);
+            p = to.find(',');
+        }
+        if (!to.empty())
+        {
+            if (idx == 0)
+                return urls::pct_string_view(
+                    to.substr(0, p));
+            --idx;
+        }
+        return boost::none;
+    }
+
+    urls::pct_string_view
+    param_or_empty(urls::string_view k) const noexcept
```

Comment:
This is like `value_or`?

Reply:
The function can't be `noexcept` because the conversion of `k` to `pct_string_view` can throw (if the encoding is invalid)